### PR TITLE
Don't consider opscode-chef-mover service status when checking ha-status

### PR DESCRIPTION
**Pull request by poliva83**
Obvious fix for https://github.com/chef/chef-server/issues/361

### CODE REVIEW

File: omnibus/files/private-chef-ctl-commands/ha.rb [hunk 1]
```diff
@@ -94,6 +94,9 @@
   ha_services = Dir.chdir(running_config['runit']['sv_dir']){Dir.glob('**/keepalive_me').map{|f| File.dirname(f)}}.sort
 
   get_all_services.sort.each do |service_name|
+    # opscode-chef-mover service is only used during an upgrade, and does not need to be running all of the time
+    next if service_name.eql? "opscode-chef-mover"
```
Comment (stevendanna) [inline omnibus/files/private-chef-ctl-commands/ha.rb]: We may want to use `omnibus-ctl`'s concept of hidden services here.  In that case, this check would become:

```
next if hidden_services.include?("opscode-chef-mover")
```

**Comment by stevendanna** [comment]
Thanks! This is a good improvement, I left a minor comment about using the hiddent_services array that omnibus-ctl already provides.

**Comment by poliva83** [comment]
@stevendanna I can change the PR to use hidden_services array. But can you please explain to me how this array becomes available to the ha.rb script?

The only reference to it I found is here:
https://github.com/chef/chef-server/blob/003067dfe6d42c9adfedd77a0dc8d69fb0f169de/omnibus/files/private-chef-cookbooks/private-chef/attributes/default.rb#L24

But to be perfectly honest I didn't really understand how ha.rb script was able to leverage the get_all_services array either.

**Comment by stevendanna** [comment]
@poliva83 Happy to try to explain.

TL;DR: You have access to any instance method in the Omnibus::Ctl class.

When we load a command definition like ha.rb, an instance of
Omnibus::Ctl eval's the relevant file:

https://github.com/chef/omnibus-ctl/blob/master/lib/omnibus-ctl.rb#L172-L174

The add_command_under_category function used to define new
commands, is actually a methods on the Omnibus::Ctl class. It
uses the block provided to add_command_under_category to define a
new instance method the Omnibus::ctl class:

https://github.com/chef/omnibus-ctl/blob/master/lib/omnibus-ctl.rb#L190

That new instance method, like any other instance method in that
class, has access to all other the other methods already defined
in Omnibus::Ctl.

Both hidden_services and get_all_services are instance methods in
the Omnibus::Ctl class

  https://github.com/chef/omnibus-ctl/blob/master/lib/omnibus-ctl.rb#L320
  https://github.com/chef/omnibus-ctl/blob/master/lib/omnibus-ctl.rb#L397

which is why you have access to them here.

**Comment by poliva83** [comment]
@stevendanna Thanks for explanation, things are more clear now. I tested and updated my pull request. Should be good now.

**Comment by stevendanna** [comment]
:+1 This looks good to merge.
